README.md

```diff
@@ -116,6 +116,23 @@ Whether the system is connected to Insights; valid values are `present`
 Whether the system is configured to run Insights remediation; valid values are
 `present` (to ensure remediation) and `absent`.
 
+    rhc_insights:
+      tags: {}
+
+A dictionary of tags that is added to the system record in Host Based Inventory (HBI).
+Used to search for hosts. This tags has keys and values. Values for each key can be any 
+data type. An example of the tags configured in the `insights-client` [documentation](https://access.redhat.com/documentation/en-us/red_hat_insights/2022/html/client_configuration_guide_for_red_hat_insights/con-insights-client-tagging-overview_insights-cg-adding-tags):
```

Comment:
If `tags` is `null` or empty, nothing is done.  Existing tags are replaced with the given tags.  If you want to remove all tags and the tags.yaml file, use `tags: {state: "absent"}`

Reply:
Hmm if empty means "no changes" too, then there is no way for the user to remove all the tags. I think a better way could be:
- `tags` is `null` -> no changes
- `tags` is `{}` -> remove all the tags (removing the file)
- `tags` is non-empty -> create/replace the file with the specified tags

Reply:
I would try to avoid to use the content of tags to check if the file needs to be removed. Although it is not a common tag there are not restrictions to setting those, so I would like to no add restriction here neither.

@ptoscano proposition is what I wanted. 
tags default value should be None in this case.